ci/lib/promote.py

```diff
@@ -234,3 +247,215 @@ def merge_forward(git_directory, push=False, parent_branch=None):
 
     # Set the branch back tot he one we started on
     checkout_branch(git_directory, starting_branch)
+
+
+def parse_version(version):
+    version_components = version.split('.')
+    major_version = 0
+    minor_version = 0
+    patch_version = 0
+    try:
+        major_version = int(version_components.pop(0))
+        minor_version = int(version_components.pop(0))
+        patch_version = int(version_components.pop(0))
+    except IndexError:
+        pass
+    return major_version, minor_version, patch_version
+
+
+def parse_release(release):
+    release_components = release.split('.')
+    major_release = 0
+    minor_release = None
+    stage = None
+    try:
+        major_release = int(release_components.pop(0))
+        minor_release = int(release_components.pop(0))
+        stage = release_components.pop(0)
+    except IndexError:
+        pass
+    return major_release, minor_release, stage
+
+
+def set_spec_version(spec_file, version, release):
+    version_regex = re.compile("^(version:\s*)(.+)$", re.IGNORECASE)
+    release_regex = re.compile("^(release:\s*)(.+)$", re.IGNORECASE)
+    in_f = open(spec_file, 'r')
+    out_f = open(spec_file + ".new", 'w')
+    for line in in_f.readlines():
+        match = re.match(version_regex, line)
+        if match:
+            line = "".join((match.group(1), version, "\n"))
+        match = re.match(release_regex, line)
+        if match:
+            line = "".join((match.group(1), release, "%{?dist}\n"))
+
+        out_f.write(line)
+
+    in_f.close()
+    out_f.close()
+    shutil.move(spec_file + ".new", spec_file)
+    print("updated %s to %s-%s" % (spec_file, version, release))
```

Comment:
You're right, but that change wasn't made in this PR so I'm going to disregard it for now. I think I'd rather just switch these scripts to work under python3, because...they already do :)

The only thing potentially blocking that is whether we run these scripts on rhel7 or not, so I'll make a note to look into this and create/fix any outstanding issues. I think that changing shebangs to python3 and ensuring we build on the fedora-np image and not, say, rhel7, is all it will take.